asdf_astropy/converters/fits/tests/test_fitswcs.py

```diff
@@ -0,0 +1,68 @@
+import asdf
+import numpy as np
+import pytest
+from astropy import wcs
+
+
+def create_sip_distortion_and_tabular_wcs():
+    rng = np.random.default_rng()
+    twcs = wcs.WCS(naxis=2)
+    twcs.wcs.crval = [251.29, 57.58]
+    twcs.wcs.cdelt = [1, 1]
+    twcs.wcs.crpix = [507, 507]
+    twcs.wcs.pc = np.array([[7.7e-6, 3.3e-5], [3.7e-5, -6.8e-6]])
+    twcs._naxis = [1014, 1014]
+    twcs.wcs.ctype = ["RA---TAN-SIP", "DEC--TAN-SIP"]
+
+    # Generate random SIP coefficients
+    a = rng.uniform(low=-1e-5, high=1e-5, size=(5, 5))
+    b = rng.uniform(low=-1e-5, high=1e-5, size=(5, 5))
+
+    # Assign SIP coefficients
+    twcs.sip = wcs.Sip(a, b, None, None, twcs.wcs.crpix)
+    twcs.wcs.set()
+
+    # Creates a WCS object with distortion lookup tables
+    img_world_wcs = wcs.WCS(naxis=2)
+    img_world_wcs.wcs.crpix = 1, 1
+    img_world_wcs.wcs.crval = 0, 0
+    img_world_wcs.wcs.cdelt = 1, 1
+
+    # Create maps with zero distortion except at one particular pixel
+    x_dist_array = np.zeros((25, 25))
+    x_dist_array[10, 20] = 0.5
+    map_x = wcs.DistortionLookupTable(
+        x_dist_array.astype(np.float32),
+        (5, 10),
+        (10, 20),
+        (2, 2),
+    )
+    y_dist_array = np.zeros((25, 25))
+    y_dist_array[10, 5] = 0.7
+    map_y = wcs.DistortionLookupTable(
+        y_dist_array.astype(np.float32),
+        (5, 10),
+        (10, 20),
+        (3, 3),
+    )
+
+    img_world_wcs.cpdis1 = map_x
+    img_world_wcs.cpdis2 = map_y
+
+    return (twcs, img_world_wcs)
+
+
+@pytest.mark.parametrize("wcs", create_sip_distortion_and_tabular_wcs())
+@pytest.mark.filterwarnings("ignore::astropy.wcs.wcs.FITSFixedWarning")
+@pytest.mark.filterwarnings(
+    "ignore:Some non-standard WCS keywords were excluded:astropy.utils.exceptions.AstropyWarning",
+)
+def test_wcs_serialization(wcs, tmp_path):
+    file_path = tmp_path / "test_wcs.asdf"
+    with asdf.AsdfFile() as af:
+        af["wcs"] = wcs
+        af.write_to(file_path)
+
+    with asdf.open(file_path) as af:
+        loaded_wcs = af["wcs"]
+        assert wcs.to_header() == loaded_wcs.to_header()
```

Comment:
```suggestion
        assert wcs.to_header() == loaded_wcs.to_header()
```

This isn't testing the distortion information for the wcses.
I tried changing this to `wcs == loaded_wcs` and this is failing in part due to the sip attribute not round-tripping.

Reply:
The `wcs == loaded_wcs` appears to be defaulting to `object.__eq__` (so will fail since `id(wcs) != id(loaded_wcs)`). This helper function could be used instead:
https://github.com/astropy/asdf-astropy/blob/fa5f1f452412fc969b79ead4a0471d38dfa3a9cc/asdf_astropy/testing/helpers.py#L105
and reports a failure for this test since the cards don't match:
```
-> assert tuple(card_a) == tuple(card_b)
(Pdb) p card_a
('A_0_0', -2.0413459666638e-06, 'SIP distortion coefficient')
(Pdb) p card_b
('A_0_0', -2.0413459666638133e-06, 'SIP distortion coefficient')
```

Reply:
Since there isn't a helper function available to check if two WCS objects are equivalent, I thought of checking the round-tripping of the WCS through its `header`.